File: CSharp/DP.Domain/Samples/Template/DkoFixing.cs
using System.Diagnostics;

namespace DP.Domain.Samples.Template
{
    public class DkoFixing : ProductFixingTemplate
    {
        protected override void FindWorkOptionLeg()
        {
            Trace.WriteLine("DKO: Find Working Option Leg");
        }
        protected override void CheckBarriers()
        {
            Trace.WriteLine("DKO: Check barries");
        }
        protected override bool RebateBarriers()
        {
            Trace.WriteLine("DKO: Rebate barries");
            return true;
        }

        protected override void FixingOptionLeg()
        {
            Trace.WriteLine("DKO: Fixing Option leg");
        }

        protected override void CheckTriggers()
        {
            Trace.WriteLine("DKO: No trigger");
        }
    }
}

File: Python/Samples/Template/DkoFixing.py
from ProductFixingTemplate import ProductFixingTemplate


class DkoFixing(ProductFixingTemplate):
    def findWorkOptionLeg(self):
        print("DKO: Find Working Option Leg!")

    def checkBarriers(self):
        print("DKO: Check barries!")

    def rebateBarriers(self):
        print("DKO: Rebate barries!")

    def fixingOptionLeg(self):
        print("DKO: Fixing Option leg!")

    def checkTriggers(self):
        print("DKO: No trigger...")
